### ISSUE TITLE
Misalignment of Edges in ReactFlow Component when Zoom Ratio Changes using scale transfrom css

### ISSUE THREAD

**Issue by jasminou38** [xyflow/xyflow#4174]
### Describe the Bug

When using the ReactFlow component and applying a zoom ratio using the transform: scale() CSS property, the edges within the flow become misaligned when the zoom ratio changes. This misalignment affects the visual presentation of the edges and can lead to a misleading representation of the flow.

Is there a viable solution to address this issue, especially when using React Flow within a parent container alongside other components that require scaling without affecting the flow?

### Your Example Website or App

https://codesandbox.io/p/sandbox/festive-poitras-x3h9nz 

### Steps to Reproduce the Bug or Issue

1- Render a ReactFlow component within a parent container.
2- Apply a zoom ratio using the transform: scale() CSS property on the parent container.
3- Change the zoom ratio dynamically, for example, by clicking a button to zoom in or out.
4- click on update Props button when the props update the edges are miss aligned what so ever.

### Expected behavior

The edges within the ReactFlow component should remain aligned correctly with the nodes, maintaining their connections and positions relative to each other.
Changing the zoom ratio should smoothly scale the entire flow without causing misalignment or distortion of edges.
and when props change the edges should be fine.

### Screenshots or Videos

![Screenshot 2024-04-17 at 21 07 05](https://github.com/xyflow/xyflow/assets/164550997/16f69354-7045-4bd4-a56b-d4980c4db785)
![Screenshot 2024-04-17 at 21 07 12](https://github.com/xyflow/xyflow/assets/164550997/93c0bf33-8e64-4fec-aa04-2b77dbcaecd9)

### Platform

- OS: [e.g. macOS, Windows, Linux] all
- Browser: [e.g. Chrome, Safari, Firefox]: all 
- Version: [e.g. 91.1] latest code

**Comment by moklick** [comment]
If the scale changes, you would need to call `updateNodeInternals` for all nodes. What is the use case for this?

**Comment by jasminou38** [comment]
Hey @moklick , thanks for getting back to me so quickly. I gave updateNodeInternals a shot, but it didn't quite solve my issues. Here's what I'm dealing with: I'm using React Flow alongside another graph. When you click on a specific part of the graph, it should show different data depending on what you clicked. Plus, I've got zoom controls to scale the graph in or out if the info's too big for the window.

Right now, I've got things sort of working by grabbing the zoomRatio and using setViewPort in React Flow. But ideally, I'd like to be able to update the data without needing to worry about zoom ratios
